include/zephyr/arch/arch_interface.h

```diff
@@ -1240,6 +1242,35 @@ bool arch_pcie_msi_vector_connect(msi_vector_t *vector,
  */
 void arch_spin_relax(void);
 
+#ifdef CONFIG_ARCH_STACKWALK
```

Comment:
Do we need this as a Kconfig?  Seems like a feature that, if unused, would be dropped from the link.  The only other reason to make it tunable would be to optimize build time for uncalled code, but again this is tiny and IMHO we don't want to be playing games like that at the arch level.

Reply:
Removed `CONFIG_ARCH_STACKWALK` but I retained the `CONFIG_ARCH_HAS_STACKWALK`, which is used to guard the `CONFIG_ARCH_STACKWALK_MAX_FRAMES`. `CONFIG_ARCH_STACKWALK_MAX_FRAMES` (like the `CONFIG_EXCEPTION_STACK_TRACE_MAX_FRAMES`) will be helpful to make sure that we dont stuck in an infinite loop, LMK what you think